### DISCUSSION TITLE
feature request: copy cursor AI UX (or do better)

### DISCUSSION THREAD

**Discussion by louis030195** [zed-industries/zed#21986] [Feature Requests]
the only thing that prevent me from fully switching to zed is that AI features are not as good as cursor UX

i have ton of perf issues with cursor but still code faster because of the AI predictions and the chat is so amazing 

great editor beside that!

PS: happy to pay whatever then, team plan etc!

**Comment by dremnik** [comment]
Not sure if seconding in comments is encouraged here but this ^^

Want so badly to just use Zed...

I find myself forced to use Cursor just for these features. For me the main one that gets used is the ⌘K for the inline editing changes. Absolutely amazing when you know the change you want but don't want to think about the syntax or logic to do it.

(Their composer feature is cool but doesn't really feel like AI is quite good enough for it yet, but that's the direction I'd be heading for the future.)

Cheers

**Reply by louis030195** [reply]
i dont use cmd k personally, only chat and predictions (composer is just too drunk, at least in my situation i write ton of rust and low level stuff so LLMs are just not great at it)

**Reply by dremnik** [reply]
Just found the feature for this -- Inline Assistant.

It's just keymapped to a strange combo (^ enter).

For anyone interested in mapping to command k:

```
  {
    "context": "Editor && mode == full",
    "bindings": {
      "cmd-k": "assistant::InlineAssist"
    }
  }
```

in the zed/keymap.json

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
